ImP/preProcessing/preProcessor.py

```diff
@@ -1,24 +1,58 @@
 import cv2 
 import numpy as np
+from skimage import data
+from skimage import exposure 
+from PIL import Image
+
 
 class PreProcessor:
+    """
+    TODO: Perhaps make a new task to develop a "main" function that takes the newly taken image and takes it through a
+     process that first runs image similarity, then the low/high contrast check, and finally adjust the image until it passes the low/high
+     contrast check.
+    """
+    image_similarity_threshold = 2.2
+
+    def __init__(self, pathToNewImage):
+        self.image = cv2.imread(pathToNewImage) # for OpenCV
+        self.scikitImage = data.imread(pathToNewImage) # for scikit
+        self.pilImage = Image.open(pathToNewImage) # for PIL
+        self.path = pathToNewImage # used to save modified images to the same location, overriding them essentially
+
+    def is_similar(self, pathToOtherImage):
+        existingImage = cv2.imread(pathToOtherImage)
+        difference = cv2.subtract(self.image, existingImage)
+        self.pilImage.close()
```

Comment:
Seems weird that the pilImage.close() step is done in this function, and not in some "tear down" function. Perhaps there should be a function just for that purpose?

Or, you could preface that instances of this class should always be used with the *with* statement to ensure proper resource cleanup.